srcjs/init_shiny.js

```diff
@@ -47,12 +47,14 @@ function initShiny() {
 
     var outputs = $(scope).find('.shiny-bound-output');
     for (var i = 0; i < outputs.length; i++) {
-      var bindingAdapter = $(outputs[i]).data('shiny-output-binding');
+      var $el = $(outputs[i]);
+      var bindingAdapter = $el.data('shiny-output-binding');
       if (!bindingAdapter)
         continue;
       var id = bindingAdapter.binding.getId(outputs[i]);
       shinyapp.unbindOutput(id, bindingAdapter);
-      $(outputs[i]).removeClass('shiny-bound-output');
+      $el.removeClass('shiny-bound-output');
+      $el.removeData('shiny-output-binding');
```

Comment:
Yes. I could have just added that one line I suppose. I changed the test to check the class instead of the data, because the former is more obviously visible when debugging using the Chrome inspector (and therefore less likely to be buggy).